### PR TITLE
executor: remove the old evaluate logic of stream agg

### PR DESCRIPTION
**Pull request by XuHuaiyu**
## What have you changed? (mandatory)
Remove the old evaluate logic of StreamAggExec.

## What is the type of the changes? (mandatory)
- Improvement (non-breaking change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)
exist test cases have covered this change

## Refer to a related PR or issue link (optional)
#6952

### CODE REVIEW

File: executor/aggfuncs/aggfuncs.go [hunk 1]
```diff
@@ -108,6 +109,24 @@ type AggFunc interface {
...
+	// | t     | a       | int(11) |
+	// +-------+---------+---------+
+	//
+	// Query: `select a, avg(a), sum(a), count(a), bit_xor(a), bit_or(a), bit_and(a), max(a), min(a), group_concat(a) from t;`
+	// +------+--------+--------+----------+------------+-----------+----------------------+--------+--------+-----------------+
+	// | a    | avg(a) | sum(a) | count(a) | bit_xor(a) | bit_or(a) | bit_and(a)           | max(a) | min(a) | group_concat(a) |
+	// +------+--------+--------+----------+------------+-----------+----------------------+--------+--------+-----------------+
+	// | NULL |   NULL |   NULL |        0 |          0 |         0 | 18446744073709551615 |   NULL |   NULL | NULL            |
+	// +------+--------+--------+----------+------------+-----------+----------------------+--------+--------+-----------------+
+	GetDefaultValue() (v types.Datum)
```
Comment (zz-jason) [inline executor/aggfuncs/aggfuncs.go]: Maybe treat this as a trait of the `AggFuncDesc` is better? Like `EvalNullValueInOuterJoin`

**Comment by XuHuaiyu** [comment]
PTAL @zz-jason

**Pull request merged**